pkg/apis/servicecatalog/types.go

```diff
@@ -519,6 +519,7 @@ type ServiceInstanceStatus struct {
 
 	// OrphanMitigationInProgress is set to true if there is an ongoing orphan
 	// mitigation operation against this ServiceInstance in progress.
+	// Deprecated: Use OrphanMitigation condition instead.
```

Comment:
I don't think that we should deprecate `OprhanMitigationInProgress`. We should continue to use that field as the primary means of determining whether an orphan mitigation is in progress. The condition is added to store the reason/message and not to supersede the field.

For supporting arguments, see https://github.com/kubernetes/kubernetes/issues/7856#issuecomment-335687733.

Reply:
@staebler initially I planned to do the same but then I realized there is little value of that:
- if OrphanMitigationInProgress is true, the OrphanMitigation condition is always set.
- as soon as OrphanMitigation condition is seen by controller, OrphanMitigationInProgress will be set to true as well, and will be reset only in case if retry timeout (a relatively rare case).

For Async operations we already have AsyncOpInProgress.
Did I miss something?

Reply:
@nilebox The kube preferred way is to use a field. I don't see a co.pelling reason here to go against the kube preferred way. The question then becomes whether there is value in having a condition for orphan mitigation. The value I see there is having a place to store the reason and message for why orphan mitigation is occurring. If that reason and message is generic, then there is no value in having an orphan mitigation condition. There is still value in not wiping out the reason and message from the Ready and Failed conditions with generic orphan mitigation reasons and messages.

Reply:
@staebler ok. do we even need to reset the `OrphanMitigationInProgress` in case of exceeding retry limit then, or just keep the flag and condition in sync?

Reply:
@nilebox Definitely keep `OrphanMitigationInProgress` in sync with the condition. I am good with the change that you have as far as keeping `OrphanMitigationInProgress` set true when the retry limit is exceeded.